lib/encodings/base64.fz

```diff
@@ -0,0 +1,155 @@
+# Base64 encoding and decoding as defined in RFC 4648
+public base64 is
+
+  # allows redefinition for base64url
+  module get_alphabet => "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/".as_codepoints.as_array
+  module get_char_62 u8 => 43 # +
+  module get_char_63 u8 => 47 # /
+  module get_encoding_name => "base64"
+
+  # NYI: compiler does currently not optimize features without arguments to treat them like fields
+  char_62_ascii := get_char_62
+  char_63_ascii := get_char_63
+  alphabet := get_alphabet
+  encoding_name := get_encoding_name
```

Comment:
no need to add a field here. Just `get_encoding_name` below.

Reply:
caching `alphabet` is much more reasonable since it is actually somewhat expensive.

Reply:
I guess you should probably define it differently though:
[(codepoint 65), (codepoint 66),...] then the compiler can easily turn this into a compile time constant.